### PR TITLE
feat: instrument main via editable data attributes

### LINKED ISSUE #72: Add data-attributes for LiveUX
We need a hidden element in the DOM with some data-attributes so we can access some data. In the future, we will require variables in the global JS workspace, but for the first golive this is the solution we need.

The data attributes can be in **any tag of the page**, like a hidden `div`.

```html
<div class="hidden" data-...></div>
```

Here are the data attributes we need.

| attribute name            | opt/mand  | type   | level   | e.g.                                                        | notes                                                                                         |
| ------------------------- | --------- | ------ | ------- | ----------------------------------------------------------- | --------------------------------------------------------------------------------------------- |
| `data-wp-version-name`    | mandatory | string | website | `"franklin"`                                                | we should have "aem" on the current website (we don’t, at the moment)                         |
| `data-wp-page-type`       | mandatory | string | page    | `"home"`, `"article"`, `"case-study"`… (names are invented) | can be the name of the template defined in Franklin, if any                                   |
| `data-wp-http-status`     | mandatory | string | page    | `"200"`, `"404" `                                           | it’s fine if it’s "static", we need just to distinguish the 404 (and other errors?) pages     |
| `data-wp-page-cookie`     | optional  | string | page    | `"accepted"`, `"C001,C002"`, `"not-accepted"`               | this should be changed using JS after the Cookie layer is dismissed                           |

This was extensively discussed in [this conversation](https://ncchat.slack.com/archives/C03K480JSF4/p1669124755714029).

### PR DESCRIPTION
**Pull request by mhaack**
Fix #72

Test URLs:

- Before: 

https://main--netcentric--hlxsites.hlx.page/who-we-are
https://main--netcentric--hlxsites.hlx.page/404.html

- After:

https://72-data-attributes-live-ux--netcentric--hlxsites.hlx.page/drafts/mblanche/who-we-are
https://72-data-attributes-live-ux--netcentric--hlxsites.hlx.page/drafts/mblanche/who-we-are-no-metadata
https://72-data-attributes-live-ux--netcentric--hlxsites.hlx.page/drafts/mblanche/who-we-are-some-metadata
https://72-data-attributes-live-ux--netcentric--hlxsites.hlx.page/404.html

### CODE REVIEW

**Comment by verlok-cn** [comment]
Hey @meryllblanchet, it looks like in the 404 page we have `data-wp-page-cookie="blah"`, will it stay like that? 
If yes, I'd rather remove the `data-wp-page-cookie` altogether.

File: 404.html [hunk 1]
```diff
@@ -26,6 +26,23 @@
...
+        if (json && json.data) {
+          const matchedRow = json.data.find((row) => (row.URL === '404.html'));
+          if (matchedRow) {
+            const main = document.querySelector('main');
+            if (main) {
+              main.setAttribute('data-wp-version-name', matchedRow['wp-version-name']);
+              main.setAttribute('data-wp-page-type', matchedRow['wp-page-type']);
+              main.setAttribute('data-wp-http-status', matchedRow['wp-http-status']);
+              main.setAttribute('data-wp-version-traffic', matchedRow['wp-version-traffic']);
+              main.setAttribute('data-wp-page-cookie', matchedRow['wp-page-cookie']);
```
Comment (verlok-cn) [inline 404.html:41]: This should not be set from metadata but from the integration with the cookie management platform (CMP), as @mhaack pointed out in #12.

**Review by verlok-cn — COMMENTED** [review]
- Please don't make the cookie value `data-wp-page-cookie` editable from editors
- Please make the default value for cookies `data-wp-page-cookie="not-accepted"`

**Comment by verlok-cn** [comment]
Anyway I wouldn't block this merge, we can go ahead and change this in the future.

**Comment by meryllblanchet** [comment]
Feature demoed to @verlok-cn and ready to go! cc @mhaack, feel free to merge.

File: 404.html (hunk 1 above)
Comment (verlok-cn) [reply 404.html:41]: As discussed with @meryllblanchet, the `main.setAttribute('data-wp-page-cookie',` row will be removed/changed when #12 will be worked.

**Pull request merged**